code/datums/status_effects/buffs.dm

```diff
@@ -366,24 +366,32 @@
 	alert_type = null
 	/// This diminishes the healing of fleshmend the higher it is.
 	var/tolerance = 1
+	/// This diminishes the healing of fleshmend if the user is cold when it is activated
+	var/freezing = FALSE
 	var/instance_duration = 10 // in ticks
 	/// a list of integers, one for each remaining instance of fleshmend.
 	var/list/active_instances = list()
 	var/ticks = 0
 
 /datum/status_effect/fleshmend/on_apply()
-	tolerance += 1
-	active_instances += instance_duration
+	check_effectiveness()
 	return TRUE
 
 /datum/status_effect/fleshmend/refresh()
+	check_effectiveness()
+	..()
+
+/datum/status_effect/fleshmend/proc/check_effectiveness()
```

Comment:
```suggestion
/datum/status_effect/fleshmend/proc/apply_new_fleshmend()
```
`check_effectiveness` suggests it will return/set a value indicating if it's effective or not. Currently it does more than just that.
The proposed rename is not the best, my tired head can't think of something better. So please rename it to your choice 😄

Reply:
I am likewise terrible at names!